src/ci/docker/dist-x86_64-linux/build-curl.sh

```diff
@@ -5,7 +5,7 @@ source shared.sh
 
 VERSION=7.51.0
 
-curl http://cool.haxx.se/download/curl-$VERSION.tar.bz2 | tar xjf -
+curl https://cool.haxx.se/download/curl-$VERSION.tar.bz2 | tar xjf -
```

Comment:
```suggestion
curl https://rust-lang-ci-mirrors.s3-us-west-1.amazonaws.com/rustc/curl-7.66.0.tar.xz | tar xJf -
```

Should probably update the version to match as well and use that (I assume it's used at some later point in the build).